packages/terra-cli/index.js

```diff
@@ -35,37 +37,39 @@ const setupCLI = () => {
-    .filter(directoryEntry => directoryEntry.isDirectory())
-    .map(directoryEntry => path.join(directoryPath, directoryEntry.name))
-);
+const subDirectoriesOfDirectory = async (directoryPath) => {
+  const currentDirectoryEnries = await fs.readdir(directoryPath, { withFileTypes: true });
+  return currentDirectoryEnries.filter(directoryEntry => directoryEntry.isDirectory())
+    .map(directoryEntry => path.join(directoryPath, directoryEntry.name));
+};
 
-const terraClIDirectoriesFromPaths = (paths, preferSrc) => (
-  paths.map(dependencyPath => path.join(dependencyPath, preferSrc ? 'src' : 'lib', 'terra-cli')).filter(directory => fs.pathExistsSync(directory))
-);
+const terraClIDirectoriesFromPaths = async (paths, preferSrc) => {
+  const cliDirectories = paths.map(dependencyPath => path.join(dependencyPath, preferSrc ? 'src' : 'lib', 'terra-cli'));
+  const pathExistsForDirectories = await Promise.all(cliDirectories.map((directory) => fs.pathExists(directory)));
+  return cliDirectories.filter((_, index) => pathExistsForDirectories[index]);
+};
 
-const loadTerraCommands = () => {
-  const { dependencies = [], devDependencies = [], name } = fs.readJSONSync(path.join(process.cwd(), 'package.json'));
-  const firstLevelDependencies = [...Object.keys(dependencies), ...Object.keys(devDependencies)];
-  const firstLevelDependencyPaths = firstLevelDependencies.map(directory => path.join(process.cwd(), 'node_modules', directory));
-  const additionalDependencyPaths = (name === 'terra-toolkit' ? subDirectoriesOfDirectory(path.join(process.cwd(), 'packages')) : []);
+const loadTerraCommandPaths = async () => {
+  const { name } = fs.readJSONSync(path.join(process.cwd(), 'package.json'));
+  const firstLevelDependencyPaths = DEPENDENCY_ALLOW_LIST.map(dependency => path.join(process.cwd(), 'node_modules', dependency));
+  const toolkitPackagePaths = (name === 'terra-toolkit' ? DEPENDENCY_ALLOW_LIST.map(dependency => path.join(process.cwd(), 'packages', dependency)) : []);
+  const currentProjectPaths = (DEPENDENCY_ALLOW_LIST.includes(name) ? [process.cwd()] : []);
   const terraCLIDirectories = [
     // Prefer lib from installed dependencies
-    ...terraClIDirectoriesFromPaths(firstLevelDependencyPaths, false),
-    // Prefer src from local dependencies
-    ...terraClIDirectoriesFromPaths([...additionalDependencyPaths, process.cwd()], true),
+    ...(await terraClIDirectoriesFromPaths(firstLevelDependencyPaths, false)),
+    // Prefer src from local mono repo packages and the current project
+    ...(await terraClIDirectoriesFromPaths([...toolkitPackagePaths, ...currentProjectPaths], true)),
   ];
-  return terraCLIDirectories.map((directory) => subDirectoriesOfDirectory(directory)).reduce((previousOutput, value) => previousOutput.concat(value), []);
+  return (await Promise.all(terraCLIDirectories.map((directory) => subDirectoriesOfDirectory(directory)))).reduce((previousOutput, value) => previousOutput.concat(value), []);
```

Comment:
totally random, but is this file getting linted? I think the linter typically throws for

```(directory) => subDirectoriesOfDirectory(directory))``` 
since you aren't using braces it'd want:
```directory => subDirectoriesOfDirectory(directory))```

Reply:
Looks like that rule got turned off a year ago: cerner/eslint-config-terra@23252cd

I did verify that file is being linted though.